lib/tasks/karma.rake

```diff
@@ -1,14 +1,22 @@
 namespace :karma  do
-  task :start => :environment do
-    with_tmp_config :start
+  task :start => :environment do |task|
+    invoke_in_test_env task.name, :start
   end
 
-  task :run => :environment do
-    with_tmp_config :start, "--single-run"
+  task :run => :environment do |task|
+    invoke_in_test_env task.name, :start, "--single-run"
   end
 
   private
 
+  def invoke_in_test_env(task_name, command, args = nil)
+    if Rails.env == 'test'
+      with_tmp_config command, args
+    else
+      system("bundle exec rake #{task_name} RAILS_ENV=test")
```

Comment:
Good thought. This would happen if you call rake without the test environment or if you use zeus. We basically start the whole bundle with all initializers a second time which is necessary to load it in the right environment. I think we should output at least a warning that this is happening so that people are aware of their mistake. Next time, they can use the `bundle exec rake karma:run RAILS_ENV=test` command straight away and save the load time of initialising the development environment.

Actually, we could just print that error with the command and exit. That is less fault tolerant, but would simplify the code and make people use the right command. Or are there cases in which we intentionally want this system call?

What do you think?